frontend/projects/upgrade/src/app/shared/services/common-export-helpers.service.ts

```diff
@@ -21,7 +21,8 @@ export class CommonExportHelpersService {
     if (data.length > 1) {
       const zip = new JSZip();
       data.forEach((element, index) => {
-        zip.file(element.name + ' (File ' + (index + 1) + ').json', JSON.stringify(element));
+        const fileName = element.name ? element.name : element.segment.name;
+        zip.file(fileName + ' (file ' + (index + 1) + ').json', JSON.stringify(element));
```

Comment:
this could be `const fileName = element.name || element.segment.name`

We should also find the segment name for the filename in the case of a single list (line 31 below). Otherwise it will always be 'undefined.json'